business/istio_config.go

```diff
@@ -542,6 +542,158 @@ func (in *IstioConfigService) GetIstioConfigDetails(ctx context.Context, namespa
 	return istioConfigDetail, err
 }
 
+// GetIstioConfigDetailsFromRegistry returns a specific Istio configuration object from Istio Registry.
+// The returned object is Read only.
+// It uses following parameters:
+// - "namespace": 		namespace where configuration is stored
+// - "objectType":		type of the configuration
+// - "object":			name of the configuration
+func (in *IstioConfigService) GetIstioConfigDetailsFromRegistry(ctx context.Context, namespace, objectType, object string) (models.IstioConfigDetails, error) {
+	var err error
+
+	istioConfigDetail := models.IstioConfigDetails{}
+	istioConfigDetail.Namespace = models.Namespace{Name: namespace}
+	istioConfigDetail.ObjectType = objectType
+
+	istioConfigDetail.Permissions = models.ResourcePermissions{
+		Create: false,
+		Update: false,
+		Delete: false,
+	}
+
+	registryCriteria := RegistryCriteria{
+		AllNamespaces: true,
+	}
+	registryConfiguration, err := in.businessLayer.RegistryStatus.GetRegistryConfiguration(registryCriteria)
+	if err != nil {
+		return istioConfigDetail, err
+	}
+	if registryConfiguration == nil {
+		return istioConfigDetail, errors.New("RegistryConfiguration is nil. This is an unexpected case. Is the Kiali cache disabled ?")
+	}
+
+	switch objectType {
+	case kubernetes.DestinationRules:
+		configs := registryConfiguration.DestinationRules
+		for _, cfg := range configs {
+			if cfg.Name == object && cfg.Namespace == namespace {
+				istioConfigDetail.DestinationRule = cfg
+				istioConfigDetail.Gateway.Kind = kubernetes.DestinationRuleType
```

Comment:
This logic is not 100% clear to external reviewers.
I understand that you iterate on the Istio registry to pick the memory config, but is the auto-generated Gateway linked with the other objects?

Perhaps is it being re-used the Gateway model just for convenience?

Reply:
oops, that is a typo